src/rationalroot.jl

```diff
@@ -0,0 +1,91 @@
+#module Tmp
+using Primes: factor
+export RationalRoot
+
+const IntorRational = Union{Integer,Rational}
+
+"""
+    RationalRoot{T<:Rational} <: Real
+
+Signed square root of a rational number
+"""
+struct RationalRoot{T<:Rational} <: Real
```

Comment:
In theory this is the perfect use case for inheriting from `AbstractIrrational`.  May also be worth reviewing discussion at https://github.com/JuliaLang/julia/issues/26550